scout/parse/omim.py

```diff
@@ -71,15 +71,41 @@ def parse_genemap2_diseases(phenotype_entry, mim_number=None):
             if term in inheritance_text:
                 inheritance.add(INHERITANCE_TERMS_MAPPER[term])
 
-        parsed_diseases.append(
-            {
-                "mim_number": disease_mim,
-                "inheritance": inheritance,
-                "description": disease_description.strip(r"?{}"),
-                "status": disease_status,
-            }
-        )
-    return parsed_diseases
+        parsed_disease_entry = {
+            "mim_number": disease_mim,
+            "inheritance": inheritance,
+            "description": disease_description.strip(r"?{}"),
+            "status": disease_status,
+        }
+        _merge_disease_entries(parsed_diseases, parsed_disease_entry)
+
+    return list(parsed_diseases.values())
+
+
+def _merge_disease_entries(parsed_diseases: dict, parsed_disease_entry: dict):
+    """Merge parsed disease entry into the current dict of parsed diseases.
+    If the old entry was already established, just append the descriptions.
+    Otherwise, the recent entry is set as the first, with the parsed disease entry status.
+    Either way, join the inheritance pattern.
+    """
+    disease_mim = parsed_disease_entry["mim_number"]
+    if disease_mim in parsed_diseases:
+        if parsed_diseases[disease_mim]["status"] == "established":
+            parsed_diseases[disease_mim]["description"] = (
+                parsed_diseases[disease_mim]["description"]
+                + ". "
+                + parsed_disease_entry["description"]
+            )
+        else:
+            parsed_diseases[disease_mim]["description"] = (
+                parsed_disease_entry["description"]
+                + ". "
+                + parsed_diseases[disease_mim]["description"]
+            )
+            parsed_diseases[disease_mim]["status"] = parsed_disease_entry["status"]
+        parsed_diseases[disease_mim]["inheritance"].update(parsed_disease_entry["inheritance"])
+    else:
```

Comment:
This condition could be at the beginning of the function, like this:

```
if disease_mim not in parsed_diseases:
        parsed_diseases[disease_mim] = entry
        return
```

So you can simplify by omitting the first if/else.

I would also rename `parsed_disease_entry` to something shorter, to make it more readable?